### ISSUE TITLE
rdrand causing amalgmation build problems

### ISSUE THREAD

**Issue by cdesjardins** [randombit/botan#61]
Using the following configure line:
```
./configure.py --disable-shared --disable-modules=mce,selftest,tls,ffi --enable-debug --via-amalgamation --disable-ssse3 --disable-sse2 --disable-avx2 --disable-aes-ni --disable-altivec
```
I get:
```
./libbotan-1.11.a(botan_all.o): In function `Intel_Rdrand':
/home/chrisd/sw/repo/botan/botan_all_internal.h:2454: undefined reference to `vtable for Botan::Intel_Rdrand'
collect2: error: ld returned 1 exit status
make: *** [botan] Error 1
make: *** Waiting for unfinished jobs....
./libbotan-1.11.a(botan_all.o): In function `Intel_Rdrand':
/home/chrisd/sw/repo/botan/botan_all_internal.h:2454: undefined reference to `vtable for Botan::Intel_Rdrand'
collect2: error: ld returned 1 exit status
make: *** [botan-test] Error 1
```

It appears that the botan_all_rdrand.cpp file is not compiled in, I have tried to play with the --single-amalgamation-file option, but it ends up with a bunch of errors like this:

```
In file included from botan_all.cpp:875:0:
/usr/lib/gcc/x86_64-linux-gnu/4.7/include/wmmintrin.h:34:3: error: #error "AES/PCLMUL instructions not enabled"
In file included from botan_all.cpp:1677:0:
/usr/lib/gcc/x86_64-linux-gnu/4.7/include/tmmintrin.h:31:3: error: #error "SSSE3 instruction set not enabled"
botan_all.cpp: In function ‘__m128i Botan::{anonymous}::aes_256_key_expansion(__m128i, __m128i)’:
botan_all.cpp:925:64: error: ‘_mm_aeskeygenassist_si128’ was not declared in this scope
botan_all.cpp: In member function ‘virtual void Botan::AES_128_NI::encrypt_n(const byte*, Botan::byte*, size_t) const’:
botan_all.cpp:1006:7: error: ‘_mm_aesenc_si128’ was not declared in this scope
botan_all.cpp:1007:7: error: ‘_mm_aesenc_si128’ was not declared in this scope
```

Just wanted to let you know, additionally I had to do add some extra #ifdefs to test_mceliece.cpp... but that is minor.

**Comment by cdesjardins** [comment]
Thanks!

**Comment by ne0h** [comment]
I'm sorry for reopening the issue:

What exactly did you include?

**Comment by randombit** [comment]
This was fixed in bdce6b1e - the 3 operations of creating the variables in order to expand the Makefile, writing build.h, and generating the amalgamation source had to be somewhat interleaved in order for everything to work
